build.fsx

```diff
@@ -25,46 +35,50 @@ module Xml =
     let value x = (x:XElement).Value
 
 module Analysis = 
-    let findAssembliesReferencing packageName = 
-        Paket.findReferencesFor packageName
-
-module Test = 
-    module NUnit = 
-        ()
-        let projects () = Analysis.findAssembliesReferencing "NUnit"
-        let run projects = 
-            projects
-            |> Seq.iter (fun p -> DotNet.test (fun o -> 
-                {o with Configuration = DotNet.BuildConfiguration.Release; NoBuild = true;
-                        Framework = if Environment.isWindows then None else Some "netcoreapp2.1"}) p)
-    module XUnit = 
-        ()
-        let projects () = Analysis.findAssembliesReferencing "xunit"
-
-        let run projects = 
-            if Environment.isWindows then
-                projects
-                |> Seq.iter (fun (p:string) -> 
-                    DotNet.test (fun o -> {o with Configuration = DotNet.BuildConfiguration.Release; NoBuild = true}) p)
-
-open AppVeyor
-open Test
+    let projectReferencing packageName project =
+        let projDoc = Xml.load project
+        let ns = projDoc |> Xml.xns
+        let packageRefName = ns |> Xml.xn2 "PackageReference"
+        let includeName = ns |> Xml.xn2 "Include"
+        let pRefs = 
+            projDoc 
+            |> Xml.descendants packageRefName
+            |> Seq.filter (fun x -> x.Attribute(includeName) <> null)
+            |> Seq.map (fun x -> x.Attribute(includeName).Value)
+        
+        if pRefs |> Seq.contains packageName then Some project
+        else None
 
 module ReleaseNotes = 
     let TickSpec = ReleaseNotes.load (__SOURCE_DIRECTORY__ </> "RELEASE_NOTES.md")
 
 module Build = 
     let rootDir = __SOURCE_DIRECTORY__
     let nuget = rootDir </> "packed_nugets"
-    let setParams (defaults :MSBuildParams) =
-        { defaults with
-            Targets = ["Rebuild"]
-            Properties =
-                [ "AllowedReferenceRelatedFileExtensions", ".pdb"
-                  "Optimize", "True"
-                  "DebugSymbols", "True"
-                  "Configuration", "Release" ]
-         }
+    let setParams (defaults :DotNet.BuildOptions) =
+        { defaults with Configuration = DotNet.BuildConfiguration.Release }
+
+module Test = 
+    module NUnit =
+        let run () =
+            !! "**/*.fsproj"
+            ++ "**/*.csproj"
+            |> Seq.choose (Analysis.projectReferencing "NUnit")
+            |> Seq.iter (fun p -> p |> DotNet.test (fun o ->
+                { o with 
+                    Configuration = DotNet.BuildConfiguration.Release
+                    NoBuild = true
+                    Framework = if Environment.isWindows then None else Some "netcoreapp2.1"}))
```

Comment:
is this condition still needed after your changes?

Reply:
No, I have removed it.